collector/netdev_common.go

```diff
@@ -11,6 +11,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//go:build !nonetdev && (linux || freebsd || openbsd || dragonfly || darwin)
```

Comment:
Hrm what is this doing?

Reply:
this is an artefact of gofmt in 1.17, should not do any harm, from release notes:

//go:build lines

The go command now understands //go:build lines and prefers them over // +build lines. The new syntax uses boolean expressions, just like Go, and should be less error-prone. As of this release, the new syntax is fully supported, and all Go files should be updated to have both forms with the same meaning. To aid in migration, gofmt now automatically synchronizes the two forms. For more details on the syntax and migration plan, see https://golang.org/design/draft-gobuild.

Reply:
Let's keep the old ones; we should move the full node exporter to go1.17 in another PR.

Reply:
gofmt works as migration assistant, it does not remove old ones, they are here as well. And it will be quite annoying for any go 1.17 users, because gofmt should be turn off, otherwise it will add this line on every save.

```go
//go:build !nonetdev && (linux || freebsd || openbsd || dragonfly || darwin)
// +build !nonetdev
// +build linux freebsd openbsd dragonfly darwin
```

Reply:
Yes, this is harmless to include.